packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+                onChangeLimit={onChangeLimit}
+                usageValue={usageValue}
+                limitValue={limitValue}
+                isEditable={isEditable}
+                limitTitle={limitTitle}
+            />
+        </div>
+    );
+};
+
+const UsageDivision = ({usageValue}: {usageValue: number}) => {
+    const isUsageRequired: boolean = !!usageValue;
+    return (
+        isUsageRequired && (
+            <div className="limit-box-usage">
+                <label className="form-control-label">Usage</label>
+                <span className="limit-box-usage-value">{usageValue}</span>
+            </div>
+        )
+    );
+};
+
+const LimitDivision = ({
+    id,
+    limitValue,
+    usageValue,
+    isEditable,
+    limitTitle,
+    onChangeLimit,
+}: {
+    id: string;
+    limitValue: number;
+    usageValue: number;
+    isEditable: boolean;
+    limitTitle: string;
+    onChangeLimit: (limit: string) => void;
+}) => {
+    const limitValueString: string = limitValue.toString();
+    const minLimitValue: number = usageValue || 0;
+    if (isEditable) {
+        return (
+            <InputConnected
+                id={id}
+                type="number"
+                labelTitle={limitTitle}
+                defaultValue={limitValueString}
+                min={minLimitValue}
+                classes="limit-box-limit form-group input-field validate"
+                onChange={(limit) => onChangeLimit(limit)}
+            ></InputConnected>
+        );
+    } else {
+        return (
+            <div className="limit-box-limit">
+                <label className="form-control-label">{limitTitle}</label>
+                <span className="limit-box-limit-value">{limitValue}</span>
+            </div>
+        );
+    }
+};
+
+const ProgressBar = ({
+    usageValue,
+    isLimitTheGoal,
+    limitValue,
+}: {
+    usageValue: number;
+    isLimitTheGoal: boolean;
+    limitValue: number;
+}) => {
+    const isUsageRequired: boolean = !!usageValue;
+    const progressClass: string = getProgressBarClass({usageValue, limitValue});
+    const progressClasses = classNames('limit-box-bar', progressClass, {
+        'mod-green': isLimitTheGoal,
+    });
+
+    return (
+        isUsageRequired && (
+            <div className="limit-box-footer">
+                <div className={progressClasses}></div>
+            </div>
+        )
+    );
+};
+
+const getProgressBarClass = ({usageValue, limitValue}: {usageValue: number; limitValue: number}): string => {
+    const progress: number = Math.round((usageValue / limitValue) * 100);
+    return progress <= 100 ? `progress-${progress.toString()}` : `progress-100`;
+};
+
+export const Limit = connect(mapStateToProps, mapDispatchToProps)(LimitDisconnect);
+
+LimitDisconnect.defaultProps = {
+    limitLabel: 'Limit',
+    limit: 100,
+    isLimitEditable: false,
+    isLimitTheGoalToReach: false,
+    isHistoryIncluded: false,
+};
+
+Limit.defaultProps = {...LimitDisconnect.defaultProps};
```

Comment:
This is interesting. but I don't get why you did that here, attaching the default props to the LimitDisconnect this way and then spreading it in the limit components looks over engineering. 
Why no a simple `const limitDefaultProps`  ?